backend/src/main/java/gov/cdc/usds/simplereport/db/repository/TestResultUploadRepository.java

```diff
@@ -1,5 +1,19 @@
 package gov.cdc.usds.simplereport.db.repository;
 
+import gov.cdc.usds.simplereport.db.model.Organization;
 import gov.cdc.usds.simplereport.db.model.TestResultUpload;
+import java.util.Date;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
-public interface TestResultUploadRepository extends AuditedEntityRepository<TestResultUpload> {}
+public interface TestResultUploadRepository extends AuditedEntityRepository<TestResultUpload> {
+  @Query(
+      "SELECT record FROM TestResultUpload record WHERE (record.organization = :org) and (cast(:startDate as date) is null or record.createdAt >= :startDate) and (cast(:endDate as date) is null or record.createdAt <= :endDate)")
+  Page<TestResultUpload> findAll(
```

Comment:
This query LGTM, but I'm curious - did something like "findAllForOrganizationBetweenDates" not work? Usually Spring is pretty smart about those queries.

Reply:
yeah that's a good question, I didn't use it because we don't always specify the date, its only there for the filter and I didn't want to create all possible query variation of filters. Another option was to use a `JpaSpecificationExecutor`, but I think its really complicated and plain jpql is simpler in this case.

Reply:
Ah I see, that makes sense. Definitely agree about the specification executor, they're powerful but probably overkill here.